apps/web/utils/logic/nextNavigation.ts

```diff
@@ -0,0 +1,143 @@
+// repo
+import { NextNavigationLogic } from "@repo/data/useWalkthroughData";
+// local
+import {
+  WalkthroughAnswerType,
+  WalkthroughStoreAnswers,
+} from "../../stores/AnswerStore";
+
+type NavigationIdType = string | undefined;
+
+export const getNextNavigationId = (
+  nextNavigationLogic: NextNavigationLogic[],
+  answerState: WalkthroughStoreAnswers,
+) => {
+  // setup nextNavigationId variable
+  let nextNavigationId: NavigationIdType;
+
+  // loop through nextNavigationLogic to find the correct next question
+  for (const navigationLogicItem of nextNavigationLogic) {
+    // if fallback item is found, set nextNavigationId to fallback item
+    // paradigms dictate that the fallback item should be the last item in the array
+    if (navigationLogicItem.nextLogicType === "fallback") {
+      nextNavigationId = navigationLogicItem.nextNavigateTo;
+      break;
+    }
+
+    // get answer to check
+    const answerToCheck = answerState[navigationLogicItem.answerToCheck || ""];
+
+    // check if answerToCheck exists
+    if (answerToCheck) {
+      // check if answerValue exists
+      if (navigationLogicItem.answerValue) {
+        /*
+         * NOTE: This logic is only for single answer value strings
+         * there is no support for multiple answers to check in this section
+         */
+
+        // check witch nextLogicType to use
+        switch (navigationLogicItem.nextLogicType) {
+          case "equals":
+            nextNavigationId = nextLogicTypeEquals(
+              answerToCheck,
+              navigationLogicItem.answerValue,
+              navigationLogicItem.nextNavigateTo,
+            );
+            break;
+          case "doesNotContain":
+            nextNavigationId = nextLogicTypeDoesNotContain(
+              answerToCheck,
+              navigationLogicItem.answerValue,
+              navigationLogicItem.nextNavigateTo,
+            );
+            break;
+        }
+      } else if (navigationLogicItem.answerValues) {
+        /*
+         * NOTE: This logic is only for answerValues arrays only
+         */
+
+        // check witch nextLogicType to use
+        switch (navigationLogicItem.nextLogicType) {
+          case "containsAny":
+            nextNavigationId = nextLogicTypeContainsAny(
+              answerToCheck,
+              navigationLogicItem.answerValues,
+              navigationLogicItem.nextNavigateTo,
+            );
+            break;
+        }
+      }
+    }
+    // TODO - if no answer to check it means its either an ane|or logic type
```

Comment:
I'm thinking me and you can norm on this. I generally write TODO's like this as well so maybe we can designate in the TODO if you or I can complete it. We could do something like `TODO: Matt` for example if it's something I'm personally tracking that I'll finish that you don't need to worry about. 

We can norm on this if you have any other suggestions.

Reply:
Good call. Because yeah, sometimes they're just notes to future me about where I see future functionality for a following ticket going.